pandas/tseries/resample.py

```diff
@@ -324,6 +326,11 @@ def _resample_timestamps(self, kind=None):
 
                 result.index = result.index + loffset
 
+        if kind == "timedelta" and len(result.index) > 0:
```

Comment:
@jreback: I had tried putting it in `_get_time_delta_bins` to avoid cluttering the `_resample_timestamp` function with type-specific code but couldn't get it working properly. I think the code that takes care of downsampling might have something to do with it.

I can try and get it working, but it'll take me a while...